akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowGraphCompileSpec.scala

```diff
@@ -227,10 +226,10 @@ class FlowGraphCompileSpec extends AkkaSpec {
 
     "build with implicits and variance" in {
       FlowGraph.closed() { implicit b ⇒
-        def appleSource = b.add(Source(PublisherProbe[Apple]))
-        def fruitSource = b.add(Source(PublisherProbe[Fruit]))
-        val outA = b add Sink(SubscriberProbe[Fruit]())
-        val outB = b add Sink(SubscriberProbe[Fruit]())
+        def appleSource = b.add(Source(TestPublisher.probe[Apple]))
```

Comment:
Hmm. It is sad that Sink is final. We can still make probe.sink work, because if you have the Module, you can create a Sink from it, since there is an akka private constructor: https://github.com/akka/akka/blob/release-2.3-dev/akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala#L76

It is not as nice though:

    x.to(Sink.publisher(probe)) // current
    x.to(probe) // If probe could be a Sink
    x.to(probe.sink) // If probe cannot be a sink, but it can still create one

What makes me wonder though is if we want to expose methods like .via(Graph[FlowShape]) and .to(Graph[SinkShape]) in the DSL. Then x.to(probe) would work and Sink could remain final.

@rkuhn WDYT?